### PR TITLE
Implement Cos IntrinsicFunction

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

File: integration_tests/intrinsics_03.f90 [hunk 1]
```diff
@@ -2,10 +2,15 @@ program intrinsics_03
 use iso_fortran_env, only: dp => real64
 real :: x
 real(dp) :: a, r1, r2
-x = cos(9.5)
-print *, x
 a = 4.2_dp
+
+x = cos(9.5)
+if (x /= -0.997172177) error stop
```
Comment (Smit-create) [inline integration_tests/intrinsics_03.f90:8]: Should we test this with: `abs(x + 0.997172177) > 1e-7`?

File: src/lfortran/pickle.cpp [hunk 2]
```diff
@@ -273,8 +273,8 @@ class ASRPickleVisitor :
             s.append(color(fg::green));
         }
         switch (x) {
-            INTRINSIC_NAME_CASE(Sin)
-            INTRINSIC_NAME_CASE(Cos)
+            INTRINSIC_NAME_CASE(sin)
+            INTRINSIC_NAME_CASE(cos)
```
Comment (czgdp1807) [inline src/lfortran/pickle.cpp:277]: Can we somehow shift this switch-case to `intrinsic_function_registry.h`?

Comment (Thirumalai-Shaktivel) [reply src/lfortran/pickle.cpp:277]: I will try to move it.

**Comment by Thirumalai-Shaktivel** [comment]
This PR is ready for review!

**Review by czgdp1807 — APPROVED** [review]

File: src/libasr/pass/intrinsic_function_registry.h [hunk 3]
```diff
@@ -35,19 +35,33 @@ typedef ASR::asr_t* (*create_intrinsic_function)(
     const std::function<void (const std::string &, const Location &)>);
 
 enum class IntrinsicFunctions : int64_t {
-    Sin,
-    Cos,
+    sin,
+    cos,
```
Comment (certik) [inline src/libasr/pass/intrinsic_function_registry.h:39]: I would follow the ASR convention to use uppercase for these enumerators, consistent with other enumerations in ASR.

**Review by certik — APPROVED** [review]

**Pull request merged**